session_test.go

```diff
@@ -1208,6 +1208,34 @@ func TestSession_sendMsg_Timeout(t *testing.T) {
 	}
 }
 
+func TestWindowOverflow(t *testing.T) {
+	// Ensures:
+	//
+	// 1. We don't accept a message that's too big.
+	// 2. We unlock after resetting the stream.
+	for i := uint32(1); i < 100; i += 2 {
+		func() {
```

Comment:
the anonymous function is just to set the scope for the defers? can probably be removed and the closes put at bottom of the loop

Reply:
I'm trying to avoid leaking anything, even on fatal errors.